### ISSUE TITLE
Self defined body class is not displayed in frontend pages anymore

### ISSUE THREAD

**Issue by bopsoid** [contao/contao#2386]
**Affected version(s)**

Contao 4.4.53 (version 4.4.32 worked, after update to 4.4.53 this error appeared)

**Description**

After updating a longer neglected Contao installation from 4.4.32 to 4.4.53 I experienced the problem, that my body classes (defined in the page layouts) were not longer shown in the source code of the front end pages.

The resulting tag now looks similar to this:

```
<body id="top" class="" itemscope itemtype="http://schema.org/WebPage">
```

Normally the class parameter should be set with my self defined body class from the page layout.

I read somewhere, that body classes that refers to different browsers (e.g. win firefox gecko fx81 etc.) were removed. Maybe the output of the body class parameter defined in page layout was accidentally removed connected to this change?

I have a work around with three different templates of fe_page.html where I hard coded my needed body classes, but that's not the nicest way to do this. ;-)

**Comment by fritzmg** [comment]
I cannot confirm this issue. What does your `fe_page` template look like? The body class is output here:

https://github.com/contao/contao/blob/234882260f1a160903e14e96222824029a25b640/core-bundle/src/Resources/contao/templates/frontend/fe_page.html5#L28

**Comment by bopsoid** [comment]
Hello,

fe_page looks like this:

`<body id="top" class="{{ua::class}}<?php if ($this->class) echo ' ' . $this->class; ?>"<?php if ($this->onload): ?> onload="<?= $this->onload ?>"<?php endif; ?> itemscope itemtype="http://schema.org/WebPage">`

So, pretty much the same.

But the class parameter I defined in the page layout is not displayed in the front end page.

This occured just after updating from 4.4.32 to 4.4.53. I didn't change any setttings or contents, it was a mere result of the update.

Source output is now class = ""

**Comment by fritzmg** [comment]
> Source output is now class = ""

Did you check the actual source or did you use the DOM inspector?

**Comment by contaoacademy** [comment]
I have tested this with Contao 4.4.53.
I get the defined body-class from layout an from page-level.

Do you use any extensions?

![classes](https://user-images.githubusercontent.com/23719181/94917772-7abf1e00-04b1-11eb-9a02-0c515994c2d3.jpg)

**Comment by bopsoid** [comment]
@fritzmg: 

I checked both and in both cases it shows class=""

@contaoacademy:

Yes, I use the following plugins:

ausi/contao-slug-backport
codefog/contao-cookiebar
codefog/contao-mobile_menu
con4gis/maps (+ all needed dependencies)
delahaye/dlh_googlemaps (parallel install to con4gis because we are in an actual change process)
heimrichhannot/contao-hyphenator-bundle
madeyourday/contao-rocksolid-antispam
madeyourday/contao-rocksolid-columns

That's it.

**Comment by bopsoid** [comment]
Just to be sure I'm not making any mistakes:

Page-Class:

![Screenshot_2020-10-02 Themes › Hospiz am Buck › Seitenlayouts › Layout ID 1 bearbeiten - Contao Open Source CMS](https://user-images.githubusercontent.com/21007947/94918864-b8bd4180-04b3-11eb-86f2-12e1f4b57f1c.png)

Layout-Class:

![Screenshot_2020-10-02 Seitenstruktur › Seite ID 5 bearbeiten - Contao Open Source CMS](https://user-images.githubusercontent.com/21007947/94918877-beb32280-04b3-11eb-8baa-630a1d0abdae.png)

Result (source code):

![Unbenannt-1](https://user-images.githubusercontent.com/21007947/94918908-c8d52100-04b3-11eb-8dd7-1b74d694ac4f.png)

**Comment by bopsoid** [comment]
Sorry, Page-Class and Layout-Class are mixed up but that doesn't make any difference. :-)

**Comment by contaoacademy** [comment]
`madeyourday/contao-rocksolid-antispam` isn't needed any more. It is implemented in core since [Contao 4.4](https://contao.org/de/news/contao_4-4-0.html)

I try to add this extensions to my test installation and give you a feedback.

**Comment by contaoacademy** [comment]
After adding the same extensions my `class` is also empty.

The problem is caused by the extension `heimrichhannot/contao-hyphenator-bundle`.

This ticket could be closed. The problem doesn't come from contao-core modules.
Please contact the creator of the extension.
